third_party/httplib/httplib.hpp

```diff
@@ -2543,7 +2543,7 @@ inline void read_file(const std::string &path, std::string &out) {
 
 inline std::string file_extension(const std::string &path) {
   Match m;
-  static auto re = Regex("\\.([a-zA-Z0-9]+)$");
+  auto re = Regex("\\.([a-zA-Z0-9]+)$");
```

Comment:
I met segfault due to static variable destruction for a few times, I usually do `absl::NoDestruct` to completely avoid destruction (it's a globally-scoped variable and don't destruct anyway)
Reference: https://github.com/abseil/abseil-cpp/blob/master/absl/base/no_destructor.h

Reply:
Absolutely, this is definitely a tradeoff for stability versus performance here. Hopefully the permanent fix in httplib will preserve both.